code/surface_area/utils.py

```diff
@@ -0,0 +1,81 @@
+from pathlib import Path
+from typing import Iterable
+import nibabel as nib
+
+
+def find_surface_files(
+    input_dir: Path, patterns: Iterable[str] | None = None
+) -> list[Path]:
+    """
+    Find surface files matching the specified patterns.
+    """
+    if patterns is None:
+        patterns = ["*mid*.surf.gii", "*mid*.rsl.gii"]
+
+    input_dir = input_dir.absolute()
+
+    input_files = []
+    for pattern in patterns:
+        input_files.extend(input_dir.rglob(f"**/{pattern}"))
+
+    # Remove duplicates
+    return list(set(input_files))
+
+
+def detect_density(n_vertices: int) -> str:
+    density = str(round(n_vertices / 1000)) + "K"
```

Comment:
```suggestion
    density = str(round(n_vertices / 1000)) + "k"
```

Should be a lowercase "k" in the name.